pkg/apis/awsproviderconfig/v1beta1/awsmachineproviderconfig_types.go

```diff
@@ -180,6 +180,10 @@ type EBSBlockDeviceSpec struct {
 	// may only be attached to machines that support Amazon EBS encryption.
 	Encrypted *bool `json:"encrypted,omitempty"`
 
+	// Indicates the KMS key that should be used to encrypt the Amazon EBS volume.
+	// The ARN for the key must be specified.
+	KMSKey AWSResourceReference `json:"kmsKey,omitempty"`
```

Comment:
Look at the API reference it suggests a Key ID can also be used? If that's possible then we should probably support it.
Alternatively, if we don't want to, or it's not, maybe this should just be `KMSKeyARN *string` instead to not give the option to specify the ID?

>// Identifier (key ID, key alias, ID ARN, or alias ARN) for a customer managed
// CMK under which the EBS volume is encrypted.

Reply:
> Looking at the API reference it suggests a Key ID can also be used? If that's possible then we should probably support it.
> https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_EbsBlockDevice.html

yeah looking at the docs, it seems like both ID and ARN styles are supported. So maybe we can support both. what the preference order between ID and ARN? I can make that change @JoelSpeed 

> Alternatively, if we don't want to, or it's not, maybe this should just be KMSKeyARN *string instead to not give the option to specify the ID?

Even if we do just want to support ARN, using the reference allows us to support filters at a later stage, when maybe AWS starts supporting that.

Reply:
A quick look through the code suggests that ID is preferred as every other use is either just the ID or ID then filter, so I'd suggest ID then ARN in this case

Reply:
@JoelSpeed fixed in 2d397b5